core/src/main/java/io/questdb/cutlass/http/WaitProcessor.java

```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cutlass.http;
+
+import io.questdb.cairo.pool.ex.EntryUnavailableException;
+import io.questdb.mp.*;
+import io.questdb.std.time.MillisecondClock;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.PriorityQueue;
+import java.util.concurrent.locks.LockSupport;
+
+public class WaitProcessor implements RescheduleContext {
```

Comment:
It feels that this class should be a job?

Reply:
Made it SynchronizedJob